sympy/controlsys/lti.py

```diff
@@ -0,0 +1,757 @@
+from __future__ import print_function, division
+
+from sympy import (
+    Symbol, var, Function, simplify, oo, exp, Eq,
+    Poly, lcm, LC, degree, Integral, integrate,
+    Matrix, BlockMatrix, eye, zeros,
+    latex, ShapeError, ImmutableMatrix, MutableMatrix,
+    SparseMatrix, MutableDenseMatrix
+)
+
+# import scipy for numercial results
+from scipy.linalg import expm
+from scipy.integrate import quad
+
+import numpy as np
+__all__ = ['StateSpaceModel', 'TransferFunctionModel']
+
+_matrixTypes = (
+    Matrix, ImmutableMatrix, MutableMatrix, SparseMatrix, MutableDenseMatrix)
+
+
+class StateSpaceModel(object):
+    """state space model (ssm) of a linear, time invariant control system
+
+    Represents the standard state-space model with state matrix A, input matrix B, output matrix C, and
+    transmission matrix D. This makes the linear controll system:
+        (1) x'(t) = A * x(t) + B * u(t);    x in R^n , u in R^k
+        (2) y(t)  = C * x(t) + D * u(t);    y in R^m
+    where u(t)  is any input signal, y(t) the corresponding output, and x(t) the systems state.
+
+    Parameters
+    ==========
+
+    arg : TransferFunctionModel, List of Sympy-Matrix
+        tfm to construct the state space model from, or the Matrices A,B,C,D in a list
+
+    See Also
+    ========
+
+    TranferFunctionModel: transfer function model of a lti system
+
+    References
+    ==========
+
+    Joao P. Hespanha, Linear Systems Theory. 2009.
+    """
+
+    def __init__(self, arg):
+
+        if isinstance(arg, TransferFunctionModel):
+
+            # call the private method for realization finding
+            self.represent = self._find_realization(arg.G, arg.s)
+
+            # create a block matrix [[A,B], [C,D]] for visual representation
+            self.BlockRepresent = BlockMatrix([[self.represent[0], self.represent[1]],
+                                               [self.represent[2], self.represent[3]]])
+            return None
+
+        else:
+            # store the argument as representation of the system
+            try:
+                self.represent = arg[:4]
```

Comment:
Would a preferable API be:

`StateSpaceModel(A, B, C, D)` instead of `StateSpaceModel([A, B, C, D])`?